system-config/qubes-usb-rules.conf

```diff
@@ -0,0 +1,8 @@
+## Allow USB hubs.  Everything else requires this.
+allow with-interface match-all { 09:00:00 09:00:01 09:00:02 }
+
+## Allow HID devices, including keyboards and mice
+allow with-interface match-all { 03:00:00 03:00:01 03:00:02 03:01:01 03:01:02 }
```

Comment:
This implements https://github.com/QubesOS/qubes-issues/issues/6959 in its current state.

While I like seeing my issue addressed, I also fear that it'll break many current installs (incl. mine).

For example I have an external sound card which would be disabled, external USB HDs or smart cards with Qubes VMs would go away etc.

I could obviously fix all of that manually, but I guess many users would be surprised.

From my point of view this should be way more permissive by default (e.g. by allowing everything connected at boot time) and well-documented for users to create their own possibly more restrictive policy and/or there should be a big red warning in the changelog.

Btw I usually use 03:*:* for HID as a more permissive example. Also nice catch with `match-all`, which I hadn't seen before.

Reply:
Good point @3hhh. I had a plan for external USB disks, but indeed there are more cases like this. So, I propose to make the whole thing conditional on `usbcore.authorised_default=0` in kernel cmdline - if not added explicitly, nothing will change. And the installer can add it when it considers it safe (for example not installing to the external USB disk), or the user can remove it back (if really needed...).

We will also include this in the upgrade instruction (to add the option if safe).